backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/CreateExperimentRunHandler.java

```diff
@@ -0,0 +1,381 @@
+
+              return InternalFuture.sequence(futureLogs, executor)
+                  .thenAccept(unused2 -> {}, executor);
+            },
+            executor);
+    // TODO .thenCompose(handle -> artifactHandler.logArtifacts(newExperimentRun.getId(),
+    // newExperimentRun.getArtifactsList()), executor)
+    // TODO .thenCompose(handle -> datasetHandler.logDatasets(newExperimentRun.getId(),
+    // newExperimentRun.getDatasetsList()), executor)
+
+    // TODO .thenCompose(handle -> featureHandler.logFeatures(newExperimentRun.getId(),
+    // newExperimentRun.getFeaturesList()), executor)
+    // TODO .thenCompose(handle -> addCodeVersionSnapShot(), executor)
+    // TODO .thenCompose(handle -> versioned_inputs, executor)
+  }
+
+  private EnvironmentBlob sortPythonEnvironmentBlob(EnvironmentBlob environmentBlob) {
+    EnvironmentBlob.Builder builder = environmentBlob.toBuilder();
+    if (builder.hasPython()) {
+      PythonEnvironmentBlob.Builder pythonEnvironmentBlobBuilder = builder.getPython().toBuilder();
+
+      // Compare requirementEnvironmentBlobs
+      List<PythonRequirementEnvironmentBlob> requirementEnvironmentBlobs =
+          new ArrayList<>(pythonEnvironmentBlobBuilder.getRequirementsList());
+      requirementEnvironmentBlobs.sort(
+          Comparator.comparing(PythonRequirementEnvironmentBlob::getLibrary));
+      pythonEnvironmentBlobBuilder
+          .clearRequirements()
+          .addAllRequirements(requirementEnvironmentBlobs);
+
+      // Compare
+      List<PythonRequirementEnvironmentBlob> constraintsBlobs =
+          new ArrayList<>(pythonEnvironmentBlobBuilder.getConstraintsList());
+      constraintsBlobs.sort(Comparator.comparing(PythonRequirementEnvironmentBlob::getLibrary));
+      pythonEnvironmentBlobBuilder.clearConstraints().addAllConstraints(constraintsBlobs);
+
+      builder.setPython(pythonEnvironmentBlobBuilder.build());
+    }
+    return builder.build();
+  }
+
+  private InternalFuture<Boolean> checkInsertedEntityAlreadyExists(ExperimentRun experimentRun) {
+    return jdbi.withHandle(
+        handle -> {
+          String queryStr =
+              "SELECT count(id) FROM experiment_run WHERE "
+                  + " name = :experimentRunName "
+                  + " AND project_id = :projectId "
+                  + " AND experiment_id = :experimentId "
+                  + " AND deleted = false ";
+
+          var query =
+              handle.createQuery(queryStr).bind("experimentRunName", experimentRun.getName());
+          query.bind("projectId", experimentRun.getProjectId());
+          query.bind("experimentId", experimentRun.getExperimentId());
+
+          long count = query.mapTo(Long.class).one();
+          return count > 0;
+        });
+  }
+
+  private String buildRoleBindingName(
+      String roleName, String resourceId, String vertaId, String resourceTypeName) {
+    return roleName + "_" + resourceTypeName + "_" + resourceId + "_" + "User_" + vertaId;
+  }
```

Comment:
This is already defined somewhere, right? As we have to use it for deletion too.

Reply:
PR: https://github.com/VertaAI/modeldb/pull/2185